app.js

```diff
@@ -0,0 +1,153 @@
+require('dotenv').config();
+
+let PORT = process.env.PORT;
+let express = require('express');
+let app = express();
+const MongoClient = require('mongodb').MongoClient;
+const MONGO_URI = process.env.MONGO_URI_MLAB;
+console.log(MONGO_URI);
+const bodyParser = require('body-parser');
+
+app.use(bodyParser.json());
+
+app.listen(PORT, () => {
+  console.log(`Server is up and running on port ${PORT}`);
+});
+
+app.get('/initialize', (req, res) => {
+  MongoClient.connect(MONGO_URI, (err, db) => {
+    if(err) {
+      // TODO handle db error
+      res.send('Error connecting to db' + err);
+    }
+
+    const users = [
+      { "userId": 1, "firstName": "Amr", "lastName": "Desouky" },
+      { "userId": 2, "firstName": "Ernie", "lastName": "Casilla" },
+      { "userId": 3, "firstName": "Ben", "lastName": "Sammons" }];
+    const accounts = [
+      { "type": "Checking", "userId": 1, "amount": 300.00, "currency": "USD", "timestamp": Date.now() },
+      { "type": "Saving",   "userId": 1, "amount": 200.00, "currency": "USD", "timestamp": Date.now() },
+      { "type": "Checking", "userId": 2, "amount": 300.00, "currency": "EUR", "timestamp": Date.now() },
+      { "type": "Saving",   "userId": 3, "amount": 300.00, "currency": "USD", "timestamp": Date.now() }];
+
+    db.collection('users').remove((error, result) => { });
+    db.collection('accounts').remove((error, result) => { });
+    db.collection('users').insert(users, (error, result) => { });
+    db.collection('accounts').insert(accounts, (error, result) => { });
+  });
+  res.send('done');
+});
+
+app.get('/account/:userID/balance', (req, res) => {
+  MongoClient.connect(MONGO_URI, (err, db) => {
+    if(err) {
+      // TODO handle db error
+      res.send('Error connecting to db');
+    }
+    // TODO input validation
+    const userId = parseInt(req.params.userID);
+    
+    getAccountBalance(db, 'accounts', userId)
+      .then(result => {
+        res.send(result);
+      })
+      .catch(error => {
+        res.send('Error getAccountBalance ' + error);
```

Comment:
Are you sure that the error thrown is type Error (for example, what does the mongo lib throw)? This will not necessarily access `.message` on the error being sent back. Depending on what error happens to be here you could be sending `Error getAccountBalance [Object object]`